README.md

```diff
@@ -24,24 +24,25 @@ from cuqi.data import grains
 from cuqi.distribution import Laplace_diff, GaussianCov 
 from cuqi.problem import BayesianProblem
 
-# Step 1: Model and data
-model, data, info = Deconvolution2D.get_components(dim=128, phantom=grains())
+# Step 1: Model and data, y = Ax
+A, y_data, info = Deconvolution2D.get_components(dim=128, phantom=grains())
 
-# Step 2: Prior
-prior = Laplace_diff(location=np.zeros(model.domain_dim),
-                     scale=0.01,
-                     bc_type='neumann',
-                     physical_dim=2)
+# Step 2: Prior, x ~ Laplace_diff(0, 0.01)
+x = Laplace_diff(location=np.zeros(A.domain_dim),
+                 scale=0.01,
+                 bc_type='neumann',
+                 physical_dim=2)
 
-# Step 3: Likelihood
-likelihood = GaussianCov(mean=model, cov=0.0036**2).to_likelihood(data)
+# Step 3: Likelihood, y ~ N(Ax, 0.0036^2)
+y = GaussianCov(mean=A@x, cov=0.0036**2)
 
-# Step 4: Posterior samples
-samples = BayesianProblem(likelihood, prior).sample_posterior(200)
+# Step 4: Set up Bayesian problem and sample posterior
+BP = BayesianProblem(y, x).set_data(y=y_data)
```

Comment:
Great comment Amal!  I didn't think about this before, but completely agree that `set_data` having to return the BP (self) is a bit confusing. The syntax
```
BP.y = y_data
``` 
would be very consistent with the single-line (*densities, **data) syntax - is this possible? If so, `set_data` is probably redundant.  

The main strength of having `set_data` I think is that it makes it clearer that we are specifying observed data for a variable, whereas `BP.y = y1` could potentially be misunderstood to be replacing the random variable `y` given to the Bayesian Problem by `y1` (whatever that might be).

Reply:
After our discussion today we decided to stick with the `set_data` method that returns self (so it can be used either in the same line or in the line below). We agreed that BP is different than JointDistribution and so it need not follow exactly the same notation. BP is the non-expert interface, so in fact it should likely behave in a more user friendly way.